A36-xds-for-servers.md

```diff
@@ -238,6 +283,11 @@ result that can be hard-coded. This applies to `destination_port` which relies
 on `use_original_dst`. It also applies to `server_names`, `transport_protocol`,
 `application_protocols` which depend on `Listener.listener_filters`.
 
+While an always-failing matcher may be pruned and ignored when matching a new
+connection, it must still be checked when verifying that the most-specific
+matching logic is guaranteed to produce a unique result during Listener
+validation.
```

Comment:
Could you please explain why we must consider always-failing matchers during the validation phase?

Reply:
> While an always-failing matcher may be pruned and ignored when matching a new connection
To be clear, we can ONLY prune a filter-chain with an always-failing matcher in the round of that matcher, and not earlier in the match algorithm. 

For example, if a filter chain is the single best match based on `prefix_ranges` but also has `application_protocols` set, the overall match should result in no filter chain being matched, which means using the `default_filter_chain` if configured or closing the connection if not configured.

Reply:
IIUC, we could have the following case:
FC1: has `prefix_range` and `application_protocols` set
FC2: has `prefix_range` set

If I end up pruning FC1 in the validation phase, then a connection which matches FC1 more specifically in `prefix_range` match could end up picking FC2. But in this case, you are saying FC1 should be picked initially since it is the most specific `prefix_range` match, but then gets dropped because it has `application_protocols` set, and thereby we end up with the default filter chain.

Reply:
> Could you please explain why we must consider always-failing matchers during the validation phase?

Because we "support" all the fields. "We look at it, so we have to verify it." The fact that we know that some of the matchers will never match doesn't change the fact that the control plane provided an invalid matcher set.

If we ever support the features that allow other fields to vary, that shouldn't change the validity of the matchers, because we were already consuming these matcher fields.